Ignore `Notes.txt` in Helm charts

**Describe the bug**
Helm charts can hold `Notes.txt` in their `templates` subfolder. If present, this file will break Cyclops, and the module won't be deployed.

**To Reproduce**
Steps to reproduce the behavior:
1. Add `Notes.txt` in the `templates` subfolder of your Helm chart 
2. Open Cyclops
3. Click on the Add New Module button and load your Helm chart
4. Try to deploy it and fail :/

**This error message might be related to the issue:**
`error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type map[string]interface {}    `


**Expected behavior**
We want Cyclops to ignore this file when deploying modules.

> I suggest you look into `cyclops-ctrl/internal/template/helm.go`. The issue is in the `mapHelmChart` function when it wants to map the "templates" section.
> 
> I believe the condition should be expanded with something like `&& parts[2] != "Notes.txt"`

@petar-cvit I modified the code with the above instructions from @KaradzaJuraj and then tried building it by referring to [CONTRIBUTING.md](https://github.com/cyclops-ui/cyclops/blob/main/CONTRIBUTING.md) but as I ran `make start`, It was then stuck at the same logs for a while then I had to `Ctrl+C` to exit.

## Here are the logs from my machine:
```
$ make start
go run -tags musl ./cmd/main
2024-04-22T14:45:37+05:30       INFO    setup   starting handler
2024-04-22T14:45:37+05:30       INFO    setup   telemetry disabled

unable to connect to redis
error bootstrapping redis dial tcp 127.0.0.1:6379: connect: connection refused
[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
 - using env:   export GIN_MODE=release
 - using code:  gin.SetMode(gin.ReleaseMode)

[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
 - using env:   export GIN_MODE=release
 - using code:  gin.SetMode(gin.ReleaseMode)

[GIN-debug] GET    /ping                     --> github.com/cyclops-ui/cycops-ctrl/internal/handler.(*Handler).pong.func1 (1 handlers)
[GIN-debug] POST   /create-config            --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).StoreConfiguration-fm (1 handlers)
[GIN-debug] GET    /create-config/:name      --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).GetConfiguration-fm (1 handlers)
[GIN-debug] GET    /configuration-details    --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).GetConfigurationsDetails-fm (1 handlers)
[GIN-debug] GET    /configuration/:name/versions --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).GetConfigurationsVersions-fm (1 handlers)       
[GIN-debug] GET    /templates                --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).GetTemplate-fm (1 handlers)
[GIN-debug] GET    /templates/initial        --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).GetTemplateInitialValues-fm (1 handlers)
[GIN-debug] GET    /templates/store          --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).ListTemplatesStore-fm (1 handlers)
[GIN-debug] PUT    /templates/store          --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).CreateTemplatesStore-fm (1 handlers)
[GIN-debug] POST   /templates/store/:name    --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).EditTemplatesStore-fm (1 handlers)
[GIN-debug] DELETE /templates/store/:name    --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Templates).DeleteTemplatesStore-fm (1 handlers)
[GIN-debug] GET    /modules/:name            --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).GetModule-fm (1 handlers)
[GIN-debug] GET    /modules/list             --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).ListModules-fm (1 handlers)
[GIN-debug] DELETE /modules/:name            --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).DeleteModule-fm (1 handlers)
[GIN-debug] POST   /modules/new              --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).CreateModule-fm (1 handlers)
[GIN-debug] POST   /modules/update           --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).UpdateModule-fm (1 handlers)
[GIN-debug] GET    /modules/:name/history    --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).GetModuleHistory-fm (1 handlers)
[GIN-debug] POST   /modules/:name/manifest   --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).Manifest-fm (1 handlers)
[GIN-debug] GET    /modules/:name/currentManifest --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).CurrentManifest-fm (1 handlers)
[GIN-debug] GET    /modules/:name/resources  --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).ResourcesForModule-fm (1 handlers)
[GIN-debug] DELETE /modules/:name/resources  --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).DeleteModuleResource-fm (1 handlers)
[GIN-debug] GET    /modules/:name/template   --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).Template-fm (1 handlers)
[GIN-debug] GET    /modules/:name/helm-template --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).HelmTemplate-fm (1 handlers)
[GIN-debug] GET    /resources/pods/:namespace/:name/:container/logs --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).GetLogs-fm (1 handlers)        
[GIN-debug] GET    /resources/pods/:namespace/:name/:container/logs/download --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).DownloadLogs-fm (1 handlers)
[GIN-debug] GET    /manifest                 --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).GetManifest-fm (1 handlers)
[GIN-debug] GET    /resources                --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Modules).GetResource-fm (1 handlers)
[GIN-debug] GET    /nodes                    --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Cluster).ListNodes-fm (1 handlers)
[GIN-debug] GET    /nodes/:name              --> github.com/cyclops-ui/cycops-ctrl/internal/controller.(*Cluster).GetNode-fm (1 handlers)
[GIN-debug] [WARNING] You trusted all proxies, this is NOT safe. We recommend you to set a value.
Please check https://pkg.go.dev/github.com/gin-gonic/gin#readme-don-t-trust-all-proxies for details.
[GIN-debug] Environment variable PORT="8888"
[GIN-debug] Listening and serving HTTP on :8888
2024-04-22T14:45:37+05:30       INFO    controller-runtime.metrics      Metrics server is starting to listen    {"addr": ":8081"}
2024-04-22T14:45:37+05:30       INFO    setup   starting manager
2024-04-22T14:45:37+05:30       INFO    Starting server {"kind": "health probe", "addr": "[::]:8082"}
2024-04-22T14:45:37+05:30       INFO    starting server {"path": "/metrics", "kind": "metrics", "addr": "[::]:8081"}
2024-04-22T14:45:37+05:30       INFO    Starting EventSource    {"controller": "module", "controllerGroup": "cyclops-ui.com", "controllerKind": "Module", "source": "kind source: *v1alpha1.Module"}
2024-04-22T14:45:37+05:30       INFO    Starting Controller     {"controller": "module", "controllerGroup": "cyclops-ui.com", "controllerKind": "Module"}
2024-04-22T14:45:37+05:30       INFO    Starting workers        {"controller": "module", "controllerGroup": "cyclops-ui.com", "controllerKind": "Module", "worker count": 1}
^C2024-04-22T16:52:07+05:30     INFO    Stopping and waiting for non leader election runnables
2024-04-22T16:52:07+05:30       INFO    shutting down server    {"path": "/metrics", "kind": "metrics", "addr": "[::]:8081"}
2024-04-22T16:52:07+05:30       INFO    Stopping and waiting for leader election runnables
2024-04-22T16:52:07+05:30       INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "module", "controllerGroup": "cyclops-ui.com", "controllerKind": "Module"}
2024-04-22T16:52:07+05:30       INFO    All workers finished    {"controller": "module", "controllerGroup": "cyclops-ui.com", "controllerKind": "Module"}
2024-04-22T16:52:07+05:30       INFO    Stopping and waiting for caches
2024-04-22T16:52:07+05:30       INFO    Stopping and waiting for webhooks
2024-04-22T16:52:07+05:30       INFO    Wait completed, proceeding to shutdown the manager
make: *** [Makefile:159: start] Error 1
```

> `make start` will start the backend part of Cyclops, which serves as a Kubernetes controller and a CRUD API. After that, you need to start the Cyclops frontend in the `cyclops-ui`, which connects to the mentioned backend. That should be it. From here, you should be able to change the code and test it on your cluster. Hope I helped. If you have any more questions, feel free to post them here.

Thank you @petar-cvit as of now I am stuck with the UI part, I ran `yarn install` inside `cyclops-ui` following `yarn start` but it throws this error due to which I'm having a hard time moving forward, I've been trying to fix it all day:

```
Failed to compile.

Module not found: Error: Can't resolve '/mnt/g/OpenSource/cyclops/Issue#203-Ignore-Notes-txt/cyclops/cyclops-ui/node_modules/webpack-dev-server/client/index.js?protocol=ws%3A&hostname=0.0.0.0&port=3000&pathname=%2Fws&logging=none&overlay=%7B%22errors%22%3Atrue%2C%22warnings%22%3Afalse%7D&reconnect=10&hot=true&live-reload=true' in '/mnt/g/OpenSource/cyclops/Issue#203-Ignore-Notes-txt/cyclops/cyclops-ui'
ERROR in Module not found: Error: Can't resolve '/mnt/g/OpenSource/cyclops/Issue#203-Ignore-Notes-txt/cyclops/cyclops-ui/node_modules/webpack-dev-server/client/index.js?protocol=ws%3A&hostname=0.0.0.0&port=3000&pathname=%2Fws&logging=none&overlay=%7B%22errors%22%3Atrue%2C%22warnings%22%3Afalse%7D&reconnect=10&hot=true&live-reload=true' in '/mnt/g/OpenSource/cyclops/Issue#203-Ignore-Notes-txt/cyclops/cyclops-ui'

webpack compiled with 1 error
Files successfully emitted, waiting for typecheck results...
Issues checking in progress...
No issues found.
```

--- a/cyclops-ctrl/internal/template/helm.go
+++ b/cyclops-ctrl/internal/template/helm.go
@@ -173,7 +173,7 @@ func (r Repo) mapHelmChart(chartName string, files map[string][]byte) (*models.T
 			continue
 		}
 
-		if len(parts) > 2 && parts[1] == "templates" && parts[2] != "Notes.txt" {
+		if len(parts) > 2 && parts[1] == "templates" && (parts[2] != "Notes.txt" && parts[2] != "NOTES.txt") {
 			manifestParts = append(manifestParts, string(content))
 			continue
 		}

--- a/cyclops-ctrl/internal/template/helm.go
+++ b/cyclops-ctrl/internal/template/helm.go
@@ -173,7 +173,7 @@ func (r Repo) mapHelmChart(chartName string, files map[string][]byte) (*models.T
 			continue
 		}
 
-		if len(parts) > 2 && parts[1] == "templates" {
+		if len(parts) > 2 && parts[1] == "templates" && parts[2] != "Notes.txt" {
 			manifestParts = append(manifestParts, string(content))
 			continue
 		}

